langstream-cli/src/main/java/ai/langstream/cli/commands/BaseCmd.java

```diff
@@ -441,6 +441,7 @@ public static File downloadHttpsFile(
                             + response.body());
         }
         Files.write(tempFile, response.body());
+        tempFile.toFile().setReadable(true, false);
```

Comment:
is this file containing secrets ? it should not be readable by anyone, but only from the user.

I also suggest to use the NIO UnixFileAttributes to perform this operation, and not the legacy File API